language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MappingContextProvider.java

```diff
@@ -187,7 +194,99 @@ protected boolean hasReadonlyKW(Node evalNodeAtCursor) {
+                .collect(Collectors.toList());
+        return this.getSpreadFieldCompletionItemList(visibleSymbols, context);
+    }
+
+    private boolean isSpreadable(Symbol symbol, List<RecordFieldSymbol> fields) {
+        Optional<TypeSymbol> typeDescriptor = SymbolUtil.getTypeDescriptor(symbol);
+        if (typeDescriptor.isEmpty()) {
+            return false;
         }
+        TypeSymbol rawType;
+        if (typeDescriptor.get().typeKind() == TypeDescKind.FUNCTION) {
+            Optional<TypeSymbol> returnTypeSymbol =
+                    ((FunctionTypeSymbol) typeDescriptor.get()).returnTypeDescriptor();
+            if (returnTypeSymbol.isEmpty()) {
+                return false;
+            }
+            typeDescriptor = returnTypeSymbol;
+        }
+
+        rawType = CommonUtil.getRawType(typeDescriptor.get());
+        if (rawType.typeKind() != TypeDescKind.RECORD) {
+            return false;
+        }
+        Map<String, RecordFieldSymbol> recordFields = ((RecordTypeSymbol) rawType).fieldDescriptors();
+        for (RecordFieldSymbol fieldSymbol : recordFields.values()) {
+            if (!isAMember(fieldSymbol, fields)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isAMember(RecordFieldSymbol field, List<RecordFieldSymbol> fields) {
+        return field.getName().isPresent() && fields.stream().anyMatch(validField ->
+                validField.getName().isPresent() && validField.getName().get().equals(field.getName().get())
+                        && field.typeDescriptor().subtypeOf(validField.typeDescriptor()));
+    }
+
+    private List<LSCompletionItem> getSpreadFieldCompletionItemList(List<Symbol> visibleSymbols,
+                                                                    BallerinaCompletionContext ctx) {
+        List<Symbol> processedSymbols = new ArrayList<>();
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        visibleSymbols.forEach(symbol -> {
+            if (processedSymbols.contains(symbol)) {
+                return;
+            }
+            Optional<TypeSymbol> typeDescriptor = SymbolUtil.getTypeDescriptor(symbol);
+            if (typeDescriptor.isPresent() && typeDescriptor.get().typeKind() == TypeDescKind.FUNCTION) {
+                typeDescriptor = ((FunctionTypeSymbol) typeDescriptor.get()).returnTypeDescriptor();
+            }
+            String typeName = (typeDescriptor.isEmpty() || typeDescriptor.get().typeKind() == null) ? "" :
+                    CommonUtil.getModifiedTypeName(ctx, typeDescriptor.get());
+            CompletionItem cItem;
+            if (symbol.kind() == FUNCTION) {
+                cItem = SpreadFieldCompletionItemBuilder.build((FunctionSymbol) symbol, typeName, ctx);
+            } else {
+                cItem = SpreadFieldCompletionItemBuilder.build(symbol, typeName);
+            }
```

Comment:
What if we do this check inside the `SpreadFieldCompletionItemBuilder` to  maintain SRP?